apiservers/portlayer/restapi/handlers/exec_handlers.go

```diff
@@ -78,6 +78,18 @@ func (handler *ExecHandlersImpl) Configure(api *operations.PortLayerAPI, netCtx
 		log.Fatalf("ExecHandler ERROR: %s", err)
 	}
 
+	vch, err := guest.GetSelf(ctx, execSession)
+	if err != nil {
+		log.Fatalf("ExecHandler unable to get VCH ref")
+	}
+
+	vchVM := vm.NewVirtualMachine(ctx, execSession, vch.Reference())
+	guestInfo, err := vchVM.FetchGuestInfo(ctx)
+	if err != nil {
+		log.Fatalf("Exechandler ERROR: %s", err)
```

Comment:
Fatalf seems heavy here.   What if there is no guest info for the vch?  If guest info is required, the check that it exists should be done at a higher level (perhaps during port layer daemon startup).  Logging an error and returning i think is fine here unless there's a really good reason to stop everything and panic here.